mlflow/utils/name_utils.py

```diff
@@ -1,4 +1,31 @@
 import random
+import sys
+import uuid
+
+# SQLite maximum primary key is 64bit (1.84467441 e19)
+_EXPERIMENT_ID_FIXED_WIDTH = 18
+
+
+def _generate_unique_integer_id():
+    """
+    Utility function for generating a random fixed-length integer
+    :param id_length: The target length of the string representation of the integer without
+                      leading zeros
+    :return: a fixed-width integer
+    """
+
+    random_int = int.from_bytes(uuid.uuid4().bytes, byteorder=sys.byteorder, signed=True)
```

Comment:
Can we use `uuid.uuid4().int` or is the current approach more random?

Reply:
I was doing some tests for signed vs. unsigned to see what the distribution of unique values might be. Based on what I'm seeing it really doesn't matter. Having a signed int doesn't give us any benefit on collision avoidance (at least for the first 100M - I'm not testing more than that due to it taking ~20 minutes to test on my laptop). I'm going to use the standard unsigned UUID().int to cast to the 128bit int and position extract the last n places.